modules/updating-eus-to-eus-upgrade-cli.adoc

```diff
@@ -0,0 +1,146 @@
+// Module included in the following assemblies:
+//
+// * updating/preparing-eus-eus-upgrade.adoc
+
+:_content-type: PROCEDURE
+[id="updating-eus-to-eus-upgrade-cli_{context}"]
+= EUS-to-EUS update using the CLI
+
+.Prerequisites
+
+* Update OpenShift CLI `oc` to the target version before each update.
+
+.Procedure
+
+. Upgrade any OLM Operators to versions that are compatible with both versions you are updating to.
+
+. Verify that all machine config pools display a status of `UPDATED` and that no machine config pool displays a status of `UPDATING`.
+To view the status of all machine config pools, run the following command:
++
+[source,terminal]
+----
+$ oc get mcp
+----
++
+.Example output
++
+Output is trimmed for clarity:
++
+[source,terminal]
+----
+NAME     CONFIG                                         	UPDATED   UPDATING
+master   rendered-master-ecbb9582781c1091e1c9f19d50cf836c       True  	  False
+worker   rendered-worker-00a3f0c68ae94e747193156b491553d5       True  	  False
+----
+
+. Change to the `eus-4.12` channel. Run the following command:
++
+[source,terminal]
+----
+$ oc adm upgrade channel eus-4.12
+----
++
+[NOTE]
+====
+The `oc adm upgrade channel` command is only present in 4.9 or later.
+
+If you receive an error message indicating that `eus-4.12` is not one of the
+available channels, this indicates that Red Hat is still rolling out 4.10 to 4.12 EUS upgrades.
+This rollout process generally takes 45-90 days starting at the GA date.
+====
++
+
+. Pause the machine config pools that you want to skip reboots on. Run the following commands:
++
+[NOTE]
+====
+You cannot pause the master pool.
```

Comment:
in my opinion is kinda not the point on whatever we _can_ or _cannot_ pause a master pool, but what kind of pools we're expected to pause, which is a workers pools for the worker nodes we intend to upgrade via the eus path. we may want to rephrase the two sentences above accordingly. 

side question: do we ever intend to pause for eus only _some_ worker pools, while leaving _others_ to upgrade the regular way in cluster undergoing an eus upgrade? @wking wdyt?

Reply:
The same concern with Evgeni.

> side question: do we ever intend to pause for eus only some worker pools, while leaving others to upgrade the regular way in cluster undergoing an eus upgrade?

That sounds like canary upgrade.

Reply:
> That sounds like canary upgrade.

slightly different perhaps? canary would be like upgrading, and then unpausing pools one by one, while in this case it would be pausing only some of the pools to begin with, which will result in some of the nodes upgrading 4.10->4.11->4.12, while others 4.10->4.12 directly. however yeah maybe this process can be referred as eus+canary as well? but is it even supported/desired upgrade strategy?

Reply:
This may be a great question for @sdodson :)

Reply:
I agree that the line between EUS-to-EUS pool pausing and canary-style pool pausing is fuzzy.  I'm agnostic about whether we explicitly point out the potential overlap, or leave that as an exercise for the reader.  I expect that leaving some compute pools unpaused during an EUS-to-EUS chain is completely fine with us, because we have CI and QE coverage for those updates with unpaused compute, and we have CI and QE coverage for EUS-to-EUS with paused compute, and I can't think of anything that would make a cluster sitting part way between those tested approaches more risky.

I'm not sure how the machine-config operator handles folks attempting to pause the control plane pool, but I expect that would block updates if it is allowed at all.

Reply:
Any of them can be left unpaused, but that runs counter to the intent of completing an upgrade as quickly as possible. The one pool which should not be paused is the master pool. I'd propose that we just say to pause all pools other than the master pool and not dive into more detail than that.

Reply:
A user can leave any pools they like unpaused.  The restriction is about  which pools are allowed to be paused, and that's "any pool except the `master` control-plane pool".  Because the machine-config operator requires that pool to update before it bumps its `versions` entry in its ClusterOperator, and the cluster-version operator requires that `versions` bump before proceeding with the rest of the OpenShift-core update.

The benefit to having an unpaused pool is that you get RHCOS and other machine-config-operator-deployed updates.  The drawback is that RHCOS and some machine-config-operator-deployed updates will require node drains and reboots.  The OpenShift-core control-plane components running on control-plane nodes have a lot of time and effort invested in making them highly-available (for most cluster flavors), so there is very little benefit in deferring drains or reboots there.  Some customer workloads may be similarly robust and be fine with rolling compute reboots.  Some customer workloads may not be as graceful in handling reboots.  And even if customer workloads gracefully handle reboots, they might  still want to pause those pools to save the time of rolling a pool `maxUnavailable: 1` at a time.